services/api-server/src/simcore_service_api_server/api/routes/solvers_jobs_getters.py

```diff
@@ -360,24 +362,25 @@ async def get_job_pricing_unit(
 
 @router.get(
     "/{solver_key:path}/releases/{version}/jobs/{job_id:uuid}/logstream",
-    response_class=StreamingResponse,
+    response_class=LogStreamingResponse,
     include_in_schema=API_SERVER_DEV_FEATURES_ENABLED,
 )
 async def get_log_stream(
     solver_key: SolverKeyId,
     version: VersionStr,
     job_id: JobID,
-    log_listener: Annotated[LogListener, Depends(LogListener)],
     webserver_api: Annotated[AuthSession, Depends(get_webserver_session)],
+    director2_api: Annotated[DirectorV2Api, Depends(get_api_client(DirectorV2Api))],
+    log_distributor: Annotated[LogDistributor, Depends(get_log_distributor)],
+    user_id: Annotated[UserID, Depends(get_current_user_id)],
 ):
     job_name = _compose_job_resource_name(solver_key, version, job_id)
-    with log_context(_logger, logging.DEBUG, "Begin streaming logs"):
-        _logger.debug("job: %s", job_name)
+    with log_context(_logger, logging.DEBUG, f"Streaming logs for {job_name=}"):
         project: ProjectGet = await webserver_api.get_project(project_id=job_id)
         _raise_if_job_not_associated_with_solver(solver_key, version, project)
-        await log_listener.listen(job_id)
-        return StreamingResponse(
-            log_listener.log_generator(),
-            media_type="application/x-ndjson",
-            background=BackgroundTask(log_listener.stop_listening),
+        log_streamer = LogStreamer(user_id, director2_api, job_id, log_distributor)
+        await log_streamer.setup()
+        return LogStreamingResponse(
+            log_streamer.log_generator(),
+            background=BackgroundTask(log_streamer.teardown),
```

Comment:
OK, what do you suggest I do instead @GitHK? I added a [test](https://github.com/bisgaard-itis/osparc-simcore/blob/5081-improve-log-streaming/services/api-server/tests/unit/api_solvers/test_api_routers_solvers_jobs_logs.py#L67) which checks that the `teardown` method is called. It doesn't have to be done in the background. What's important here is that the method is called after the `log_streamer` returns.

Reply:
I am not suggesting any changes. If this works for you it's fine. Just take note on how this works and it might created unexpected side effects.
I also think in this situation it should be fine.

Reply:
I am not sure I get the gist of the problem here. is that _background_ is part of the response to the client? would it be possible to hide that part of the response maybe?

Reply:
Perfect, thanks a lot for the feedback